src/Sylius/Bundle/AdminBundle/Action/ShipShipmentAction.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\AdminBundle\Action;
+
+use Sylius\Bundle\AdminBundle\Shipper\OrderShipmentShipperInterface;
+use Sylius\Component\Core\Repository\ShipmentRepositoryInterface;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpFoundation\Session\Session;
+use Symfony\Component\HttpKernel\Exception\HttpException;
+use Symfony\Component\Security\Csrf\CsrfToken;
+use Symfony\Component\Security\Csrf\CsrfTokenManagerInterface;
+use Webmozart\Assert\Assert;
+
+final class ShipShipmentAction
+{
+    /** @var ShipmentRepositoryInterface */
+    private $shipmentRepository;
+
+    /** @var CsrfTokenManagerInterface */
+    private $csrfTokenManager;
+
+    /** @var OrderShipmentShipperInterface */
+    private $orderShipmentShipper;
+
+    public function __construct(
+        ShipmentRepositoryInterface $shipmentRepository,
+        CsrfTokenManagerInterface $csrfTokenManager,
+        OrderShipmentShipperInterface $orderShipmentShipper
+    ) {
+        $this->shipmentRepository = $shipmentRepository;
+        $this->csrfTokenManager = $csrfTokenManager;
+        $this->orderShipmentShipper = $orderShipmentShipper;
+    }
+
+    public function __invoke(Request $request): Response
```

Comment:
```suggestion
    public function __invoke(Request $request, string $id): Response
```

No need for these lines then:
```
$shipmentId = $request->attributes->get('id');
Assert::notNull($shipmentId);
```